Add CPU slow test job

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73748

This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

davidberard98 committed:
Enable CPU device_type tests on CUDA slow machines

There are no CPU-only slow test machines, so this enables device_type slow tests (which previously would never run)

[ghstack-poisoned]

davidberard98 committed:
Update on "Enable CPU device_type tests on CUDA slow machines"


There are no CPU-only slow test machines, so this enables device_type slow tests (which previously would never run)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34628803).

davidberard98 commented:
@albanD regarding slowTest on CPU, previously some of the CPU slowTests weren't running (e.g. TestNNCOpInfoCPU).

You can see that the runtime increased for the slow job:
On this PR (2.5 hrs): https://github.com/pytorch/pytorch/runs/5414065481?check_suite_focus=true
On a recent commit on master (1.5 hrs): https://github.com/pytorch/pytorch/runs/5424674750?check_suite_focus=true

Note: this also enables the CPU tests on `slow-gradcheck` jobs, should that be the case or would we only want to run it on the `slow` jobs?

albanD commented:
Why does this change the slow gradcheck tests?

davidberard98 commented:
@albanD not entirely sure what the slow-gradcheck tests do... but it seems like they're a subset of slow tests, in that `PYTORCH_TEST_WITH_SLOW=1, PYTORCH_TEST_SKIP_FAST=1` are set. So if there were any slow-gradcheck tests marked @slowTest and which use instantiate_device_type_tests(), I think we'd want to enable the CPU versions? (otherwise they would never run, since there's no cpu-only slow-gradcheck jobs, either)

davidberard98 committed:
Update on "Enable CPU device_type tests on CUDA slow machines"


There are no CPU-only slow test machines, so this enables device_type slow tests (which previously would never run)

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34628803).

davidberard98 commented:
actually, I'm going to disable the CPU tests on slow-gradcheck for now, since otherwise it times out.

malfet reviewed (changes requested):
Hmm, it feels like a much better solution would be to add slow CPU tests, wouldn't it? (which can be done by one-line change to https://github.com/pytorch/pytorch/blob/master/.github/scripts/generate_ci_workflows.py

davidberard98 committed:
Update on "Enable CPU device_type tests on CUDA slow machines"


There are no CPU-only slow test machines, so this enables device_type slow tests (which previously would never run)

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 committed:
Update on "Add CPU slow_test jobs"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34628803).

davidberard98 committed:
Update on "Add CPU slow_test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 commented:
#74398 - tests that were failing which had never run on a clang build

davidberard98 committed:
Update on "Add CPU slow_test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 committed:
Update on "Add CPU slow_test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 committed:
Update on "Add CPU slow_test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 committed:
Update on "Add CPU slow_test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 committed:
Update on "Add CPU slow_test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

davidberard98 committed:
Update on "Add CPU slow test job"


This adds CPU-only slow test jobs, which previously would never run.

Includes fixes for fx slow tests which were previously failing, but which never ran.

Differential Revision: [D34628803](https://our.internmc.facebook.com/intern/diff/D34628803)

[ghstack-poisoned]

malfet commented on .github/workflows/trunk.yml:
```diff
@@ -127,6 +127,25 @@ jobs:
       ANDROID_SIGN_PASS: ${{ secrets.ANDROID_SIGN_PASS }}
       SCRIBE_GRAPHQL_ACCESS_TOKEN: ${{ secrets.SCRIBE_GRAPHQL_ACCESS_TOKEN }}
 
+  linux-bionic-py3_7-clang9-slow-build:
+    name: linux-bionic-py3.7-clang9-slow
+    uses: pytorch/pytorch/.github/workflows/_linux-build.yml@master
+    with:
+      build-environment: linux-bionic-py3.7-clang9-slow
+      docker-image-name: pytorch-linux-bionic-py3.7-clang9
```
Do you need a separate build because there are no trunk-only builds?
@suo with new lint workflow is there a way to add a trunk-only test-shard to PR workflow?

suo replied:
it's just standard GH stuff now, so a job with an if statements would work.

Although: can we not just use one of the many trunk gcc builds instead? Is there something about clang that is special here

davidberard98 replied:
@suo how would the job with if statements work? e.g. where to put if statements?
re: gcc builds, I think we just want a cpu-only build. AFAICT all the other builds in trunk.yml are for cuda (other than parallelnative, which seems like a special configuration).

suo replied:
oh, that's interesting. Yeah it looks like there is no trunk-only CPU build. idk, personally I think this change is fine; I'd rather just do an extra build than make the workflows more marginally more complicated.

suo replied:
for reference, it would look like the same thing you have, with one additional line: 
```
  if: github.event == 'push'
```

But like I said, I feel like we should just keep pull jobs in pull.yml, and trunk jobs in trunk.yml. CPU builds take sub-10m and are super cheap, so it's not a big deal to overbuild a bit

davidberard98 commented on test/test_unary_ufuncs.py:
```diff
@@ -910,6 +910,7 @@ def test_helper(min, max):
     @onlyCPU
     @slowTest
     @dtypes(torch.float)
+    @unittest.skipIf(True, "Insufficient memory on linux.(2|4)xlarge")
```
@suo @malfet any opinions on whether we should skip this test entirely, or move the slow test to a 9xlarge just to have enough memory for this test?

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34628803).
